admin-dev/themes/new-theme/js/components/auto-complete-search.ts

```diff
@@ -29,26 +29,35 @@
  * functions which are, of course, overridable.
  */
 
+type DisplayFunction = (item: any) => string;
 interface TypeaheadJQueryDataset extends Twitter.Typeahead.Dataset<any> {
-  dataLimit: number;
+  display: string | DisplayFunction;
   value: string;
+  limit: number;
+  dataLimit: number;
+  templates: any;
+}
+
+interface TypeaheadJQueryOptions extends Twitter.Typeahead.Options {
+  minLength: number,
+  highlight: boolean,
+  hint: boolean,
   onSelect: (
```

Comment:
@NeOMakinG do you think I should create a dedicated type for this function (like I did with `DisplayFunction`) Btw I wondered if this type is then accessible globally from anywhere (and potentially could have conflicts with other types with the same name). Or is it scoped to this file only?

Reply:
I don't think so, I like the fact to have little interfaces directly inside components they aim

Reply:
Ok, I agree but I still defined such types in cases where I needed "multi types" Like `DisplayFunction | undefined`
In this case the inline function along with a pending `| undefined` would look weird

Reply:
Yeah and in this case, it looks clearer to me